### PR TITLE
Added redis_exporter module

### PR DESCRIPTION
**Pull request by vide**
It installs https://github.com/oliver006/redis_exporter which
seems to be the most used redis exporter for Prometheus.
It uses prometheus::daemon although I had to redefine the archive
extraction code in the exporter class due to packaging differences.

### CODE REVIEW

File: manifests/redis_exporter.pp [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+#  [*service_name*]
+#  Name of the node exporter service (default 'redis_exporter')
+#
+#  [*user*]
+#  User which runs the service
+#
+#  [*version*]
+#  The binary release version
+class prometheus::redis_exporter (
+  $addr                 = $::prometheus::params::redis_exporter_redis_address,
```
Comment (bastelfreak) [inline manifests/redis_exporter.pp:84]: can you add puppet4 datatypes to the parameters?

File: manifests/redis_exporter.pp [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+  $service_ensure       = 'running',
+  $service_name         = 'redis_exporter',
+  $user                 = $::prometheus::params::redis_exporter_user,
+  $version              = $::prometheus::params::redis_exporter_version,
+) inherits prometheus::params {
+
+  $release = "v${version}"
+
+  $real_download_url = pick($download_url, "${download_url_base}/download/${release}/${package_name}-${release}.${os}-${arch}.${download_extension}")
+  validate_bool($purge_config_dir)
```
Comment (bastelfreak) [inline manifests/redis_exporter.pp:114]: you can drop these validate_* calls, they're deprecated and replaced by datatypes

**Comment by bastelfreak** [comment]
ping @vide

**Comment by vide** [comment]
Sorry @bastelfreak still on 3.8 at work so I have to setup Puppet 4 to write and test the code, and currently have very little spare time :(

**Comment by bastelfreak** [comment]
Duplicated by https://github.com/voxpupuli/puppet-prometheus/pull/157

**Comment by allangood** [comment]
Hi @vide and @bastelfreak ,

I've created this PR #224 to use "redis.file" parameter as default for this exporter.

Hope this could helps!
Regards.

**Comment by bastelfreak** [comment]
Hi,
I'm going to close this PR due to inactivity. Please reopen and rebase it, if you're still interested in it.

**Pull request closed without merging**